src/main/java/de/tum/cit/aet/artemis/exercise/repository/StudentParticipationRepository.java

```diff
@@ -1361,4 +1361,48 @@ SELECT MAX(pr.id)
                 ORDER BY p.student.firstName ASC
             """)
     Page<FeedbackAffectedStudentDTO> findAffectedStudentsByFeedbackId(@Param("exerciseId") long exerciseId, @Param("feedbackIds") List<Long> feedbackIds, Pageable pageable);
+
+    /**
+     * Retrieves the logins of students affected by a specific feedback detail text in a given exercise.
+     *
+     * @param exerciseId The ID of the exercise for which affected students are requested.
+     * @param detailText The feedback detail text to filter by.
+     * @return A list of student logins affected by the given feedback detail text in the specified exercise.
+     */
+    @Query("""
+                SELECT DISTINCT p.student.login
+                FROM ProgrammingExerciseStudentParticipation p
+                LEFT JOIN p.submissions s
+                LEFT JOIN s.results r
+                LEFT JOIN r.feedbacks f
```

Comment:
Why are we using `LEFT JOIN` here? Because `exerciseId` cannot be null and `detailText`should not be null. So, I think an inner join should be more efficient here.